lib/msal-browser/src/interaction_handler/RedirectHandler.ts

```diff
@@ -2,55 +2,57 @@
+import { StringUtils, AuthorizationCodeModule, TemporaryCacheKeys, TokenResponse } from "msal-common";
 import { AuthCallback } from "../app/PublicClientApplication";
 import { BrowserConfigurationAuthError } from "../error/BrowserConfigurationAuthError";
 import { BrowserStorage } from "../cache/BrowserStorage";
 import { BrowserConstants } from "../utils/BrowserConstants";
+import { BrowserUtils } from "../utils/BrowserUtils";
 
-export class RedirectHandler extends IInteractionHandler {
+export class RedirectHandler extends InteractionHandler {
 
     private authCallback: AuthCallback;
+    private navigateToLoginRequestUrl: boolean;
 
-    constructor(authCodeModule: AuthorizationCodeModule, storageImpl: BrowserStorage,  redirectCallback: AuthCallback) {
+    constructor(authCodeModule: AuthorizationCodeModule, storageImpl: BrowserStorage, redirectCallback: AuthCallback, navigateToLoginRequestUrl: boolean) {
         super(authCodeModule, storageImpl);
         if (!redirectCallback) {
             throw BrowserConfigurationAuthError.createRedirectCallbacksNotSetError();
         }
         this.authCallback = redirectCallback;
+        this.navigateToLoginRequestUrl = navigateToLoginRequestUrl;
     }
 
     /**
      * Redirects window to given URL.
      * @param urlNavigate 
      */
-    showUI(authRequest: AuthenticationParameters): void {
+    showUI(requestUrl: string): Window {
         this.browserStorage.setItem(TemporaryCacheKeys.ORIGIN_URI, window.location.href);
-        this.authModule.createLoginUrl(authRequest).then((urlNavigate) => {
-            this.browserStorage.setItem(BrowserConstants.INTERACTION_STATUS_KEY, BrowserConstants.INTERACTION_IN_PROGRESS);
-            // Navigate if valid URL
-            if (urlNavigate && !StringUtils.isEmpty(urlNavigate)) {
-                this.authModule.logger.infoPii("Navigate to:" + urlNavigate);
-                window.location.assign(urlNavigate);
-            } else {
-                this.authModule.logger.info("Navigate url is empty");
-                throw BrowserAuthError.createEmptyRedirectUriError();
-            }
-        });
+        this.browserStorage.setItem(BrowserConstants.INTERACTION_STATUS_KEY, BrowserConstants.INTERACTION_IN_PROGRESS);
+        // Navigate if valid URL
+        if (!StringUtils.isEmpty(requestUrl)) {
+            this.authModule.logger.infoPii("Navigate to:" + requestUrl);
+            BrowserUtils.navigateWindow(requestUrl);
+        } else {
+            this.authModule.logger.info("Navigate url is empty");
+            throw BrowserAuthError.createEmptyRedirectUriError();
+        }
+        return window;
     }
 
     /**
      * Handle authorization code response in the window.
      * @param hash 
      */
-    async handleCodeResponse(locationHash: string, navigateToLoginRequestUrl?: boolean): Promise<void> {
+    async handleCodeResponse(locationHash: string): Promise<TokenResponse> {
```

Comment:
How is this different in Popup and Redirect use cases?  Can we have a base class that addresses these? This comment as well as anything with restructure can be addressed later and not be a blocker for your alpha release. I am adding for traction purposes.

Reply:
The redirect performs the navigateToLoginRequestUrl navigation if the setting is there. Popup doesn't use this. That is the main difference, we can optimize this in the future.